atlp-rwanda/stackup2-barefoot-backend
# #171380748 Social login google and facebook

#### What does this PR do?

- It allows the users to login using their google and Facebook accounts

#### Description of Task to be completed?

- It allows the user to login using their Facebook and Google accounts

#### How should this be manually tested?

- git clone https://github.com/Stackup-Rwanda/stackup2-barefoot-backend.git

- cd stackup2-barefoot-backend

- git fetch ft-login-with-fb-and-google-171380748

- git checkout ft-login-with-fb-and-google-171380748

- npm install

-  create a project on [developers facebook](https://developers.facebook.com/) and [developers google](https://developers.google.com/identity/sign-in/web/sign-in) in order to get the ID and Secret keys 

- Use http://localhost:3000/auth/google/callback as the callback URL for google authentication

- Use http://localhost:3000/auth/facebook/callback as the callback URL for Facebook authentication

- create .env file and use .env.example file for reference of the environment variables

- npm run dev

- to test google auth use this URL in your browser: http://localhost:3000/auth/google/

- to test Facebook auth use this URL in your browser: http://localhost:3000/auth/facebook/
The result should be an object of the logged-in user

#### Any background context you want to provide?

- Please remember to create a project on [developers facebook](https://developers.facebook.com/) and [developers google](https://developers.google.com/identity/sign-in/web/sign-in) in order to get the ID and Secret keys, because these keys are the ones you will use to fill in the environment variables that are in .env.example file.
#### What are the relevant pivotal tracker stories?

- [171380748](https://www.pivotaltracker.com/story/show/171380748)

#### Screenshots (if appropriate)
![facebook](https://user-images.githubusercontent.com/37122177/75885229-d3044380-5e2e-11ea-871a-ddde4e62ed0d.PNG)
![google-1](https://user-images.githubusercontent.com/37122177/75885232-d4ce0700-5e2e-11ea-9c6b-cbf55e7f030c.PNG)
![google-2](https://user-images.githubusercontent.com/37122177/75885235-d5669d80-5e2e-11ea-8ebf-3514494ca9d7.PNG)
